modules/gdnative/pluginscript/pluginscript_script.cpp

```diff
@@ -0,0 +1,407 @@
+// Godot imports
+#include "core/os/file_access.h"
+// PluginScript imports
+#include "pluginscript_instance.h"
+#include "pluginscript_script.h"
+
+#if DEBUG_ENABLED
+#define __ASSERT_SCRIPT_REASON "Cannot retrieve pluginscript class for this script, is you code correct ?"
+#define ASSERT_SCRIPT_VALID()                \
+	{                                        \
+		ERR_EXPLAIN(__ASSERT_SCRIPT_REASON); \
+		ERR_FAIL_COND(!this->can_instance()) \
+	}
+#define ASSERT_SCRIPT_VALID_V(ret)                  \
+	{                                               \
+		ERR_EXPLAIN(__ASSERT_SCRIPT_REASON);        \
+		ERR_FAIL_COND_V(!this->can_instance(), ret) \
+	}
+#else
+#define ASSERT_SCRIPT_VALID()
+#define ASSERT_SCRIPT_VALID_V(ret)
+#endif
+
+void PluginScript::_bind_methods() {
+}
+
+#ifdef TOOLS_ENABLED
+
+void PluginScript::_placeholder_erased(PlaceHolderScriptInstance *p_placeholder) {
+	placeholders.erase(p_placeholder);
+}
+
+#endif
+
+bool PluginScript::can_instance() const {
+	bool can = this->_valid || (!this->_tool && !ScriptServer::is_scripting_enabled());
+	return can;
+}
+
+// TODO: not working properly yet (_refbase is never set...)
+Ref<Script> PluginScript::get_base_script() const {
+	if (this->_refbase.ptr()) {
+		return Ref<PluginScript>(this->_refbase);
+	} else {
+		return Ref<Script>();
+	}
+}
+
+StringName PluginScript::get_instance_base_type() const {
+	if (this->_refbase.is_valid())
+		return this->_refbase->get_instance_base_type();
+	return StringName();
+}
+
+void PluginScript::update_exports() {
+	// TODO
```

Comment:
Yup, that's pretty important. :P

Reply:
I'm not sure what should be done here, so I would prefer to keep it this way for the current PR ;-)